test/shared/src/main/scala/zio/test/PrettyPrint.scala

```diff
@@ -38,46 +53,62 @@ ${indent(body.mkString(",\n"))}
 
       case product: Product =>
         val name    = product.productPrefix
+        val size    = product.productArity
         val labels0 = labels(product)
-        val body = labels0
-          .zip(product.productIterator)
-          .map { case (key, value) =>
-            s"${(key + " =").faint} ${PrettyPrint(value)}"
-          }
-          .toList
-          .mkString(",\n")
-        val isMultiline  = body.split("\n").length > 1
-        val indentedBody = indent(body, if (isMultiline) 2 else 0)
```

Comment:
This wasn't great. We were building a String with `\n` as separator to then split it twice on this same character. I implemented an optimised version of this algorithm